azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -410,6 +409,8 @@ public int selectAndUpdateExecution(final int executorId, final boolean isActive
             final ExecutableFlow exFlow =
                 ExecutableFlow.createExecutableFlowFromObject(
                     GZIPUtils.transformBytesToObject(data, encType));
+            Status status = Status.fromInteger(rs.getInt(4));
+            exFlow.setStatus(status);
```

Comment:
Maybe add an explanation in the comments for why the `status` specifically is overwritten by the value from DB column.

Reply:
done

Reply:
This is still bug prone. I see that there are 4 instances of such calls. Why not pass in the status as a parameter to createExecutableFlowFromObject and encapsulate the logic within? This way, any new future call will remain consistent.

Reply:
I made the change, thanks.